src/ui/component/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react'
+import type { FC } from 'react'
+import classNames from 'classnames'
+import * as RTooltip from '@radix-ui/react-tooltip'
+import './tooltip.scss'
+
+type TooltipProps = Omit<RTooltip.TooltipPortalProps, 'content'> & {
+  content: JSX.Element
+  trigger: JSX.Element
+  contentClassName?: string
+  triggerClassName?: string
+  side?: RTooltip.TooltipContentProps['side']
+  sideOffset?: RTooltip.TooltipContentProps['sideOffset']
+  container?: RTooltip.TooltipPortalProps['container']
```

Comment:
Please, make props generic without letting know that, for now, we are using Radix UI.

Details of implementation should not be part of this contract.

Keep in mind that this component could be part of a future design system as it stands, using Radix UI or from scratch.

Reply:
Component has been updated according to your feedback.

We will take it into account when designing future components.

Thanks!